java/src/test/java/vw/learner/VWFloatArrayLearnerTest.java

```diff
@@ -109,4 +117,64 @@ private void writeVwModelToDisk() {
     private VWFloatArrayLearner rehydrateModel() {
         return VWLearners.create("-i " + model + " -t --quiet");
     }
+
+    @Test
+    public void testCBExplore() throws IOException {
+        String[] cbTrain = new String[]{
+            "1:2:0.4 | a c",
+            "3:0.5:0.2 | b d",
+            "4:1.2:0.5 | a b c",
+            "2:1:0.3 | b c",
+            "3:1.5:0.7 | a d"
+        };
+
+        VWFloatArrayLearner vw = VWLearners.create("--quiet --cb_explore 4");
+        float[][] trainPreds = new float[cbTrain.length][];
+        for (int i=0; i<cbTrain.length; ++i) {
+            trainPreds[i] = vw.learn(cbTrain[i]);
+        }
+        float[][] expectedTrainPreds = new float[][]{
+            new float[]{0.962500f, 0.012500f, 0.012500f, 0.012500f},
```

Comment:
When performing either `--cb_explore` or `--cb_explore_adf` the user is requesting a score over actions.  On the Java side this is represented by a `float[]` where the index is the action and the value is the score.  The user would feed this array into another function that makes a decision, like the [Decision Service](https://mwtds.azurewebsites.net/).  This does bring up an interesting point though and that is that I assume that on the C side the number of actions is the same as the size of the `ACTION_SCORE::action_scores` `v_array` and that each index is filled in with a score.  @JohnLangford can you please comment as to whether or not this is correct.